### PR TITLE
Feat/search bar

### PR DESCRIPTION
**Pull request by danielzairin**
Search bar built with [cmdk](https://github.com/pacocoursey/cmdk)

https://github.com/user-attachments/assets/a9484882-5305-49d3-b66e-bafc64defa27

### CODE REVIEW

**Comment by irfancoder** [comment]
Can we add a prop to open the search results on input focus? I think this will be a nice UX improvement

File: packages/react/src/components/search-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+      },
+    },
+    defaultVariants: {
+      size: DEFAULT_SIZE,
+    },
+  },
+);
+export const SearchBarResultsDropdown: FunctionComponent<
+  ComponentProps<"div">
+> = ({ className, ...props }) => {
```
Comment (irfancoder) [inline packages/react/src/components/search-bar.tsx:217]: Rename to ~~SearchBarResultsDropdown~~ `SearchBarResults`

**Comment by danielzairin** [comment]
> Can we add a prop to open the search results on input focus? I think this will be a nice UX improvement

I went with a slightly different approach and added an `open` prop to the `SearchResults` subcomponent. This lets the developers themselves control exactly when the results dropdown should open or close.

**Review by irfancoder — APPROVED** [review]
lgtm!

**Comment by paanjoe** [comment]
> Search bar built with [cmdk](https://github.com/pacocoursey/cmdk)
> 
>  Kapture.2024-11-28.at.11.55.08.mp4

boekkkkkk

**Pull request merged**